lib/stdlib/src/filename.erl

```diff
@@ -857,7 +857,7 @@ climb(T, [_|Acc]) ->
 %% known by the code manager, i.e. code:which/1 should succeed.
 %%
 %% Rules describes how the source directory should be found given
-%% the directory for the object code.  Each rule is on the form
+%% the directory for the module.  Each rule is on the form
```

Comment:
It's not obvious that "module" here means "a module's compiled representation".  If you hate the term "object code" so much (though it is traditional), then perhaps "compiled module" or something like that would be better.

Reply:
No, my comment concerns all your changes.  Basically I oppose substituting "module" for "object code" because "module" is a highly context-dependent term (it could be the Erlang language concept, a source file, a BEAM file (or other compiled representation, the BEAM is not the only Erlang VM), or the VM's internal representation).  Even "object code" isn't ideal since the usual meaning of that is different from what's intended here.  Either "compiled module" if you want to keep things generic and not too tied to the BEAM, or "BEAM file" if you want to keep things simple and concrete, would be better.

Reply:
@mikpe so I went with "module bytecode", as an attempt to qualify the module, and because we are talking about what is loaded into memory (or even referring to the binary directly). The only exception is in the filename docs, because they are already using module without qualifying it.

Reply:
@josevalim I don't like the term "bytecode". It is technically correct when referring to the contents of the BEAM files, but it is not correct to refer to the loaded code as bytecode (it is called threaded code). Bytecode interpreters have a reputation for being slow, so I think we should avoid that term.